packages/jellyfish-transaction-builder/src/txn/txn_builder_masternode.ts

```diff
@@ -0,0 +1,35 @@
+import { BigNumber } from '@defichain/jellyfish-json'
+import { OP_CODES, Script, TransactionSegWit, CreateMasternode, ResignMasternode } from '@defichain/jellyfish-transaction'
+import { P2WPKHTxnBuilder } from './txn_builder'
+
+export class TxnBuilderMasternode extends P2WPKHTxnBuilder {
+  /**
+   * Build create masternode transaction
+   *
+   * @param {CreateMasternode} createMasternode transaction to create
+   * @param {Script} changeScript to send unspent to after deducting the (converted + fees)
+   * @return {Promise<TransactionSegWit>}
+   */
+  async create (createMasternode: CreateMasternode, changeScript: Script, network = 'regtest'): Promise<TransactionSegWit> {
```

Comment:
I think with this https://github.com/DeFiCh/jellyfish/pull/534/files#r690875965,
we need to handle `force default` logic here either based on `height` or user input(creator of the `createMasternode` msg). and add some safety checks based on that.

Reply:
i don't get it.. don't we handle it `optional`?? why do `force default` logic here??
can you submit a suggestion code here??

Reply:
I think we can safely omit this, for me it makes sense to prioritise accurate deserialization of historic (pre Eunos Paya) tx, for indexing, but why add so much complexity to enforce sending of historic tx? the practical use case is non existant

Reply:
This is actually to ensure the safety of the future, not the past. consider this scenario. someone tries to create a createMasternode tx after post eunosPaya without timelock. with the current dftx layer implementation, it would not write `0x0000` when serializing. two options

1. we let the creator of the `createMasternode` message take full responsibility, i.e. always mention timelock 0 when creating post eunosPaya. if he does not, too bad it would be the incorrect format. 
2. or else we can ask for another parameter from the user, `forceDefaultTimelock` so that he is informed, it's going to be post eunosPaya format.

I would prefer the second option.
@canonbrother if you still need code suggestions, I could provide one if we could agree that my point is valid and should be addressed.

Reply:
I see your point. I was referring to the extra logic like adding a height check. If we default to enforcing new behaviour as you suggested i.e. #2 then I'm ok

Reply:
@surangap  hmm.. understand.. i asked the suggestion to allow me better understand the request
the suggestion:
```ts
createMasternode.timelock = createMasternode.timelock ?? 0x0000
```

Reply:
@canonbrother not quite, he's suggesting adding a default value `forceDefaultTimelock` (defaults to true) to enable this behaviour

Reply:
@monstrobishi  @surangap I think this is good for me.. better ux at least... :sweat_smile: 
i will add this
```ts
createMasternode.timelock = createMasternode.timelock ?? 0x0000
```

Reply:
ok, this addresses the problem I think. we can agree that in real life nobody is going to use this to create masternodes for pre eunosPaya unless the chain is rolled back. for regtest, it's fine IMO

Reply:
out of topic.. never see a blockchain rollback its structure.. but switching network is yes (eg: eth2)